site/content/blog/simplifying-package-authoring.md

```diff
@@ -0,0 +1,301 @@
+---
+title: "Simplifying Package Authoring"
+slug: simplifying-package-authoring
+date: 2022-08-17
+author: Rohit Aggarwal
+excerpt: "Simplifying Package Authoring with kctrl"
+image: /img/logo.svg
+tags: ['carvel', 'package', 'author', 'kctrl', 'create']
+---
+
+In Carvel, Kubernetes(K8s) manifest are distributed and consumed via the concept of Package. A package Author encapsulates, version, and distribute K8s manifest for package consumers to install on a Kubernetes cluster. A package author can choose to create a package by using third party manifest e.g. they can choose to create a package from cert-manager, dynatrace etc. or they can distribute their own project K8s manifest by creating a Package.
+
+Package Authoring is an iterative process and below are the steps authors generally go through:
+1. Authors know about the K8s manifest they want to package.
+2. Add/change the manifest by adding additional overlay/template and test the package. This is the iterative part where authors want to make the changes and test them quickly.
+3. Once all the manifest are in place, create the imgpkg bundle(to be mentioned in the package) and the package itself.
+4. Add the package to the package repository for distribution.
+
+Today, package authors are supposed to know all the Carvel tools as they are being used in the package authoring journey. Learning the Carvel tools before package authoring has a learning curve off itself. As part of simplifying package authoring, we wanted to reduce this learning curve so that authors can focus on package authoring. As part of this, few commands have been added to the already existing `kctrl` cli.
```

Comment:
```suggestion
Today, package authors are supposed to know all the Carvel tools ([ytt](https://carvel.dev/ytt/), [imgpkg](https://carvel.dev/imgpkg/), [kbld](https://carvel.dev/kbld/), [vendir](https://carvel.dev/vendir/), [secretgen-controller](https://github.com/vmware-tanzu/carvel-secretgen-controller), [kapp](https://carvel.dev/kapp/), and [kapp-controller](https://carvel.dev/kapp-controller/) as they are being used in the package authoring journey. Learning the Carvel tools before package authoring has a learning curve of itself. As part of simplifying package authoring, we wanted to reduce this learning curve so that authors can focus on package authoring. As part of this, few commands have been added to the already existing `[kctrl](https://carvel.dev/blog/kctrl-release-blog/)` CLI.
```

Reply:
package authors are supposed to know all the Carvel tools? Is this meant to be 'package authors using Carvel tools are expected to know all the Carvel tools'?? 

Because, in general, package authors don't need to know all the Carvel tools if they use something else. Unless I'm missing something!

Reply:
And the last bit, I'm a bit confused. Suggested change:

As part of simplifying package authoring, we wanted to reduce this learning curve so that authors can focus on package authoring by adding a few commands to the already existing `[kctrl](https://carvel.dev/blog/kctrl-release-blog/)` CLI.

Reply:
+1

Reply:
Changed it to: `We wanted to introduce a set of CLI commands that guide users in performing most common packaging steps.`